### PR TITLE
add more bytes cmd

### PR DESCRIPTION
**Pull request by WindSoilder**
# Description

Add the following bytes command:

1. ~bytes to-str~
2. bytes at
3. bytes index-of

# Tests

Make sure you've done the following:

- [X] Add tests that cover your changes, either in the command examples, the crate/tests folder, or in the /tests folder.
- [X] Try to think about corner cases and various ways how your changes could break. Cover them with tests.
- [X] If adding tests is not possible, please document in the PR body a minimal example with steps on how to reproduce so one can verify your change works.

Make sure you've run and fixed any issues with these commands:

- [X] `cargo fmt --all -- --check` to check standard code formatting (`cargo fmt --all` applies these changes)
- [X] `cargo clippy --workspace --features=extra -- -D warnings -D clippy::unwrap_used -A clippy::needless_collect` to check that you're using the standard code style
- [X] `cargo test --workspace --features=extra` to check that all the tests pass

### CODE REVIEW

File: crates/nu-command/src/bytes/index_of.rs [hunk 1]
```diff
@@ -0,0 +1,142 @@
...
+
+#[derive(Clone)]
+pub struct BytesIndexOf;
+
+impl Command for BytesIndexOf {
+    fn name(&self) -> &str {
+        "bytes index-of"
+    }
+
+    fn signature(&self) -> Signature {
```
Comment (fdncred) [inline crates/nu-command/src/bytes/index_of.rs:29]: I'm wondering if we should have an `--all` parameter that searched the entire binary input that is passed to it. This would be super helpful for finding multi-page offsets  in TIFF files or TAR files. For instance, if every TIFF page in a multi-page TIFF starts with 0x[123] and then we return all of them, it would then be easy to store that list/array and iterate through it, doing something to each item.

File: crates/nu-command/src/bytes/to_str.rs [hunk 2]
```diff
@@ -0,0 +1,118 @@
...
+        self.column_paths.take()
+    }
+}
+
+#[derive(Clone)]
+pub struct BytesToStr;
+
+impl Command for BytesToStr {
+    fn name(&self) -> &str {
+        "bytes to-str"
```
Comment (fdncred) [inline crates/nu-command/src/bytes/to_str.rs:24]: I'm sure that `to-str` was in my gist defining byte commands but now that I see it, it makes me wonder how it's different from `decode utf-8`? If there is some difference, or we intend to make it different somehow in the future, then we should keep it. If there is no difference, I'm not sure we should keep it.

**Comment by fdncred** [comment]
oh, I'm loving these byte commands. Thanks so much for taking them on @WindSoilder. I added a couple comments and tweaked your usage text on one of the commands.

File: crates/nu-command/src/bytes/to_str.rs (hunk 2 above)
Comment (WindSoilder) [reply crates/nu-command/src/bytes/to_str.rs:24]: There is just a little difference, `to-str` only support to `utf-8`, and it support passing column path, so we can use `to-str` like this:
```
[[ColA ColB ColC]; [0x[61 73 64 66] 0x[67 68 69 66] 0x[71 77 65 72]]] | bytes to-str ColA ColC
```

But I think `decode` is more feature rich, I'm ok to remove it

File: crates/nu-command/src/bytes/index_of.rs (hunk 1 above)
Comment (WindSoilder) [reply crates/nu-command/src/bytes/index_of.rs:29]: Ah, thank you, that's a good suggestion, I'll try it

File: crates/nu-command/src/bytes/to_str.rs (hunk 2 above)
Comment (fdncred) [reply crates/nu-command/src/bytes/to_str.rs:24]: maybe we can add the column path passing to decode as well, and then we wouldn't need this?

File: crates/nu-command/src/bytes/index_of.rs (hunk 1 above)
Comment (WindSoilder) [reply crates/nu-command/src/bytes/index_of.rs:29]: Sorry, I have a simple question, for the following case
```
0x[11 11 11 11] | bytes index-of 0x[11 11]
```

Should we return:
1. [0 1 2], or
2. [0 2]?

Comment (fdncred) [reply crates/nu-command/src/bytes/index_of.rs:29]: I think the offsets should be 0 and 2 for [11 11], once we search a range I don't think we should search it again.

Comment (WindSoilder) [reply crates/nu-command/src/bytes/index_of.rs:30]: Ok, that make sence :)

**Comment by fdncred** [comment]
Nice. Thanks for the changes.
